pkg/dockerregistry/server/simplelookupicsp.go

```diff
@@ -41,7 +47,18 @@ func (s *simpleLookupICSP) FirstRequest(
 		return []reference.DockerImageReference{ref.AsRepository()}, nil
 	}
 
-	imageRefList, err := s.alternativeImageSources(ref, icspList.Items)
+	idmsList, err := s.idmsClient.List(ctx, metav1.ListOptions{})
+	if err != nil {
+		klog.Errorf("unable to list IDMS config: %s", err)
+		return []reference.DockerImageReference{ref.AsRepository()}, nil
+	}
+
+	if len(icspList.Items) > 0 && len(idmsList.Items) > 0 {
+		klog.Errorf("error finding both icsp and idms resources at the same time: %s", err)
```

Comment:
Can we clarify this message a bit? It caused me some confusion the way it's currently phrased.
What about:
```
error: found both ICSP and IDMS resources, but only one or the other is supported
```
Additionally, you're adding the error itself to the formatting, but at this point the error will be nil.